### ISSUE TITLE
[Feature request] Allow 0 for start of the video, but disable the rest

### ISSUE THREAD

**Issue by tuurep** [timmontague/youtube-disable-number-seek#3]
Hello, I would use this addon if it wasn't for the fact that I very frequently do use <kbd>0</kbd> to go to the start of the video / restart. Are you interested in making a setting to exclude disabling <kbd>0</kbd>?

**Comment by NguyenASang** [comment]
[youtube-disable-number-seek.zip](https://github.com/timmontague/youtube-disable-number-seek/files/10471732/youtube-disable-number-seek.zip)

This one does that but I don't know how to pack to .xpi (you can load unpacked this folder)

**Comment by KneeCappo** [comment]
dont worry i got it fixed, found out how to bring it over to opera GX

**Comment by soulseller** [comment]
The Home key goes to the start when you have the video on focus.

**Comment by NguyenASang** [comment]
This one will disable those keys.

[youtube-disable-number-seek.zip](https://github.com/timmontague/youtube-disable-number-seek/files/10752407/youtube-disable-number-seek.zip)

**Comment by tuurep** [comment]
Hi, sorry for replying quite late. Thanks for all the help.

I didn't know that Home key works too, that's a pretty good compromise, but so dependent on keyboard layout, I prefer to use the 0 especially because (thank you very much) Nguyen's version allows that.

I had a bit of trouble trying to add the contents of that zip to Firefox, but eventually succeeded by submitting it for manual review at [addons.mozilla.org](https://addons.mozilla.org)

This process is so cumbersome for trying out such tiny code changes :( If someone knows how to sign addons myself, or allow to test them unsigned, would like to know more :)

However it's inspiring to see how easy it is to change this code to include/exclude whichever keys you want.

~~I'll put the addon that Mozilla manually signed for me, here:
https://addons.mozilla.org/firefox/downloads/file/4073371/0b65f879fbe34df5b153-1.1.xpi~~
(Edit: link broken)

This is the same as the first .zip by NguyenASang

If you want to verify, you can right click on the link, Save Link As..., then unzip the .xpi file.

Thanks a lot, this fits my need already. If I learn how to code user settings for Firefox addons, I'd be interested in sending a pull request, or if we are content now, closing the issue as resolved is also fine by me.

Lets say, if there's no activity on this issue for a year, I'll just go ahead and close :smiley:

**Comment by soulseller** [comment]
Unfortunately, I am unable to download the signed addon from the provided link.

**Comment by tuurep** [comment]
Thanks for letting me know, I tried my old link, I can download it but actually can't unzip it, doesn't seem reliable.

Would this work instead?

~~https://github.com/tuurep/files/raw/main/disable-yt-numberkeys-except-0.xpi~~
Edit: now on addons.mozilla.org:
https://addons.mozilla.org/en-US/firefox/addon/disable-yt-numberkeys-except-0/

**Comment by soulseller** [comment]
Besides the description still says "Disables the YouTube 0-9 keyboard shortcuts which seek to different times on a video", I think it works as expected.

**Comment by tuurep** [comment]
Yea, the naming gets too mixed up if you wanna actually use this.

Good news, I updated all the names and descriptions, and was approved to host this directly on addons.mozilla.org:

https://addons.mozilla.org/en-US/firefox/addon/disable-yt-numberkeys-except-0/

**Comment by soulseller** [comment]
You could also safely drop the permissions for invidio.us because it no longer operates on this domain.

**Comment by tuurep** [comment]
Thanks, I dropped that. I want to try to fix it for third-party frontends like the invidious instances. So now I have two goals for this addon, after which I would consider it finished:

1. Allow using with any YouTube frontends' domains
* [SponsorBlock](https://github.com/ajayyy/SponsorBlock) does that so study its code
2. Add the toggle to enable/disable <kbd>0</kbd>
* This may lead me to rename the fork yet again :smiley: 

I now made a repo so it's easier to maintain and follow:
https://github.com/tuurep/disable-yt-numberkeys-except-0
